raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -211,6 +216,70 @@ contract MonitoringService is Utils {
         );
     }
 
+    function isAllowedToMonitor(
+        TokenNetwork token_network,
+        uint256 channel_identifier,
+        address closing_participant,
+        address non_closing_participant,
+        address monitoring_service_address
+    )
+        internal
+        returns (bool)
+    {
+        TokenNetwork.ChannelState channel_state;
+        uint256 settle_block_number;
+        (settle_block_number, channel_state) = token_network.getChannelInfo(
+            channel_identifier, closing_participant, non_closing_participant
+        );
+        require(channel_state == TokenNetwork.ChannelState.Closed);
+
+        // We don't actually know when the channel has been closed. So we'll
+        // make a guess so that assumed_close_block >= real_close_block.
+        uint256 assumed_settle_timeout = token_network.settlement_timeout_min();
+        uint256 assumed_close_block = settle_block_number - assumed_settle_timeout;
+        return block.number >= firstBlockAllowedToMonitor(
+            assumed_close_block,
+            assumed_settle_timeout,
+            closing_participant,
+            non_closing_participant,
+            msg.sender
+        );
+    }
+
+    function firstBlockAllowedToMonitor(
+        uint256 closed_at_block,
+        uint256 settle_timeout,
+        address participant1,
+        address participant2,
+        address monitoring_service_address
+    )
+        public pure
+        returns (uint256)
+    {
+        // avoid overflows when multiplying with percentages
+        require(settle_timeout < uint256(2**256 - 1) / 100);
```

Comment:
To prevent an overflow in `best_case_block`, I'd check
```
require(closed_at_block < uint256(2**256 - 1) / 100);
require(settle_timeout < uint256(2**256 - 1) / 100 - closed_at_block);
```

Reply:
> require(closed_at_block < uint256(2**256 - 1) / 100);

I was assuming that block numbers don't grow that large. I'll add it to make that explicit.

> require(settle_timeout < uint256(2**256 - 1) / 100 - closed_at_block);

I don't see a need for adding `closed_at_block`:
```
best_case_block
= closed_at_block + BEST_CASE * settle_timeout / 100
<= closed_at_block + 100 * settle_timeout / 100
= closed_at_block + settle_timeout
< uint256(2**256 - 1) / 100 + uint256(2**256 - 1) / 100
= uint256(2**256 - 1) / 50
< uint256(2**256 - 1)
```

Reply:
Right. We know `uint256(2**256 - 1) / 100` is a small number!